tinygrad/lazy.py

```diff
@@ -285,7 +285,9 @@ def elementwise_op(op:Union[UnaryOps, BinaryOps], *srcs:LazyBuffer, arg:Optional
 class _Device:
   def __init__(self) -> None:
     self._buffers: List[str] = [x.stem[len("ops_"):].upper() for x in (pathlib.Path(__file__).parent/"runtime").iterdir() if x.stem.startswith("ops_")]
-    self.DEFAULT: str = functools.reduce(lambda val, ele: ele if getenv(ele) == 1 else val, self._buffers, "CPU")
+    try: default = "METAL" if self["METAL"] else "CUDA" if self["CUDA"] else "GPU" if self["GPU"] else "CPU"
+    except ImportError: default = "CPU"
```

Comment:
I think this logic is not quite right - if metal dependencies are not  installed it will go to CPU instead of CUDA. I'll fix this tomorrow.